roofit/roofitcore/src/RooDataHist.cxx

```diff
@@ -1142,6 +1142,49 @@ RooPlot *RooDataHist::plotOn(RooPlot *frame, PlotOpt o) const
   return RooAbsData::plotOn(frame,o) ;
 }
 
+
+void RooDataHist::weights(double* output, RooSpan<double const> xVals, bool correctForBinSize) 
+{
+  auto nEvents = xVals.size();
+
+  RooAbsBinning const& binning = *_lvbins[0];
+
+  if (binning.isUniform()) {
+    double minVal = binning.binLow(0);
+    double maxValFirstBin = binning.binHigh(0);
+
+    double oneOverBinWidth;
+    if (correctForBinSize) {
+      oneOverBinWidth = 1. / (maxValFirstBin - minVal);
+    } else {
+      oneOverBinWidth = 1.;
+    }
```

Comment:
```suggestion
    oneOverBinWidth = correctForBinSize ? 1. / (maxValFirstBin - minVal) : 1.;
```
You can do this in a one-liner.

Reply:
And then you can also declare it const to prevent inadvertent modifications of the value.